src/Interpreters/Cache/Metadata.h

```diff
@@ -12,6 +12,48 @@ class CleanupQueue;
 using CleanupQueuePtr = std::shared_ptr<CleanupQueue>;
 
 
+struct CacheMetadata : public std::unordered_map<FileCacheKey, KeyMetadataPtr>, private boost::noncopyable
```

Comment:
Nothing is changed here, I just moved it before the `KeyMetadata` class to be able to write `friend void CacheMetadata::doCleanup()`